eZ/Publish/Core/Repository/URLService.php

```diff
@@ -106,27 +115,31 @@ public function updateUrl(URL $url, URLUpdateStruct $struct)
      */
     public function loadById($id)
     {
-        if ($this->repository->hasAccess('url', 'view') !== true) {
+        $url = $this->buildDomainObject(
+            $this->urlHandler->loadById($id)
+        );
+
+        if (!$this->permissionResolver->canUser('url', 'view', $url)) {
             throw new UnauthorizedException('url', 'view');
         }
 
-        return $this->buildDomainObject(
-            $this->urlHandler->loadById($id)
-        );
+        return $url;
     }
 
     /**
      * {@inheritdoc}
      */
     public function loadByUrl($url)
     {
-        if ($this->repository->hasAccess('url', 'view') !== true) {
+        $APIUrl = $this->buildDomainObject(
```

Comment:
```suggestion
        $apiUrl = $this->buildDomainObject(
```

or rename argument name to e.g. `$urlString`, `$urlValue`